### ISSUE TITLE
why the following autograd.grad() return None? 

### ISSUE THREAD

**Issue by lovecoffeeandmilk** [pytorch/pytorch#45405]
class func(torch.autograd.Function):

    '''
    The size of time slots is 1 bigger than the depth
    '''
    @staticmethod
    def forward(ctx, x0, w):
        out = torch.sum(x0 * w)
        ctx.save_for_backward(w, x0)
        return out

    @staticmethod
    def backward(ctx, grad_output):
        w, x0 = ctx.saved_tensors
        out = torch.sum(x0 * w)
        out.requires_grad = True
        x0.requires_grad = True
        w.requires_grad = True
        print("XXXX ", out.requires_grad, x0.requires_grad, w.requires_grad)
        dfdx = torch.autograd.grad(outputs=out, inputs=x0, grad_outputs=grad_output, allow_unused=True)
        dfdw = torch.autograd.grad(outputs=out, inputs=w, grad_outputs = grad_output , allow_unused=True)
        print("YYYY ", dfdx, dfdw)
        return dfdx, dfdw

 class Net(nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.w = torch.rand(2, 3)
    def forward(self, x):
        out = func.apply(x, self.w)
        return out

 def main():
    net = Net()
    data = torch.rand(2, 3)
    data.requires_grad = True
    gt = 5
    out = net(data)
    loss = out - gt
    print("run backward")
    loss.backward()

 

My question: why the output is None from autograd.grad() function? 

run backward
XXXX  True True True
YYYY  (None,) (None,)

**Comment by gchanan** [comment]
please ask questions on the forum: https://discuss.pytorch.org/.
